x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/results/TextEmbeddingResults.java

```diff
@@ -59,6 +61,40 @@ public TextEmbeddingResults(StreamInput in) throws IOException {
         );
     }
 
+    public static TextEmbeddingResults of(List<? extends InferenceResults> results) {
+        List<Embedding> embeddings = new ArrayList<>(results.size());
+        for (InferenceResults result : results) {
+            if (result instanceof org.elasticsearch.xpack.core.ml.inference.results.TextEmbeddingResults embeddingResult) {
+                embeddings.add(new Embedding(convertEmbeddingResultToEmbedding(embeddingResult)));
+            } else if (result instanceof org.elasticsearch.xpack.core.ml.inference.results.ErrorInferenceResults errorResult) {
```

Comment:
I'm not sure that this is correct. I see that we are getting this other type of TextEmbeddingResults from the pytorch process, but then we are converting these results from doubles to floats, which could cause a loss of precision. However, this TextEmbeddingsResults uses the Float class, so we would have to change that to avoid the loss of precision. 

Overall, this whole conversion here which involves creating a lot of new objects seems pretty inefficient, but I can't think of how to do this in a more efficient manner. I think we would probably need to make changes to the TextEmbeddingResults objects.

Reply:
yes it does appear to be doing a lot of work. 

One change is to use an array of (lowercase) floats rather than a list of Floats. We never expect the Float to be null so might as well use floats. Surprisingly this can be done without changing the serialisation code as `StreamOutput::writeFloatArray` is the functionally the same as `StreamOutput.writeCollection(values, StreamOutput::writeFloat)`, both write the size of the array/list then that number of floats. That can be done in a separate PR